.github/workflows/ci.yaml

```diff
@@ -61,10 +61,11 @@ jobs:
       - uses: actions/setup-go@v2
         with:
           go-version: v1.17
-      - uses: golangci/golangci-lint-action@v2
-        with:
-          only-new-issues: true
-          args: --timeout=5m
+      - name: Install golangci-lint
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
```

Comment:
If we're going to do this, we should pin to a specific version. However, I'm wondering if it would be possible to make sure everyone is running the same lint version, both locally and in CI. Would you be willing to see if you could do something similar to https://github.com/kcp-dev/kcp/blob/467fc8a2fc2f8b67139eba965301eae8990cb463/Makefile#L24-L27 and https://github.com/kcp-dev/kcp/blob/467fc8a2fc2f8b67139eba965301eae8990cb463/Makefile#L59-L60?

Reply:
I can give it a try, this is grabbing (technically) the same version that CI was using but I think pinning to a version would be the safest. Would it be fine to stuff away the `golangci-lint` install in the make target for linting (install if it isn't already)? I could easily drop this in the `bin/` directory so that we install the linter locally within the repo but this may not be ideal. :cat: open to ideas

Reply:
What I linked above does the install into `hack/tools`, if you follow that model.

Reply:
I see, sounds good! I will give that a try.

Reply:
wowee, you were way ahead of me :) made the requested changes
```
❯ make lint
GOBIN=/home/jbpratt/git/kcp/hack/tools ./hack/go-install.sh github.com/golangci/golangci-lint/cmd/golangci-lint golangci-lint v1.44.2
go: creating new go.mod: module fake/mod
go get: installing executables with 'go get' in module mode is deprecated.
	To adjust and download dependencies of the current module, use 'go get -d'.
	To install using requirements of the current module, use 'go install'.
	To install ignoring the current module, use 'go install' with a version,
	like 'go install example.com/cmd@latest'.
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
go get: added 4d63.com/gochecknoglobals v0.1.0
go get: added github.com/Antonboom/errname v0.1.5
go get: added github.com/Antonboom/nilnil v0.1.0
go get: added github.com/BurntSushi/toml v1.0.0
go get: added github.com/Djarvur/go-err113 v0.0.0-20210108212216-aea10b59be24
go get: added github.com/Masterminds/semver v1.5.0
go get: added github.com/OpenPeeDeeP/depguard v1.1.0
go get: added github.com/alexkohler/prealloc v1.0.0
go get: added github.com/ashanbrown/forbidigo v1.3.0
go get: added github.com/ashanbrown/makezero v1.1.0
go get: added github.com/beorn7/perks v1.0.1
go get: added github.com/bkielbasa/cyclop v1.2.0
go get: added github.com/blizzy78/varnamelen v0.6.0
go get: added github.com/bombsimon/wsl/v3 v3.3.0
go get: added github.com/breml/bidichk v0.2.2
go get: added github.com/breml/errchkjson v0.2.3
go get: added github.com/butuzov/ireturn v0.1.1
go get: added github.com/cespare/xxhash/v2 v2.1.2
....
/home/jbpratt/git/kcp/hack/tools/golangci-lint-v1.44.2 run --timeout=10m ./...

❯ make lint
/home/jbpratt/git/kcp/hack/tools/golangci-lint-v1.44.2 run --timeout=10m ./...
```

Reply:
Cool!